main.go

```diff
@@ -1,5 +1,73 @@
 package main
 
-func main() {
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"sort"
+)
+
+type HumanDecoding interface {
+	Decode(data []byte) ([]Person, []Place)
+}
+
+type Logging interface {
+	Println(v ...interface{})
+	Fatalf(format string, v ...interface{})
+}
+
+func NewHumanDecoder(service *Service) *HumanDecoder {
+	return &HumanDecoder{service: service}
+}
+
+func NewService(log Logger) *Service {
+	return &Service{log: log}
+}
+
+func (h *HumanDecoder) Decode(data []byte) ([]Person, []Place) {
+	var b Base
+	var persons []Person
+	var places []Place
+
+	err := json.Unmarshal(data, &b)
+	if err != nil {
+		log.Fatal(err)
+	}
 
+	for _, val := range b.Things {
+		data := val.(map[string]interface{})
+		if data["name"] != nil {
+			person := Person{data["name"].(string), data["age"].(float64)}
+			persons = append(persons, person)
+		} else {
+			place := Place{data["city"].(string), data["country"].(string)}
```

Comment:
i would always suggest to do type assertion with second return value check (i mean `ok`). It would prevent panics

Reply:
got it